### PR TITLE
Use native API instead of Vimscript

### PR DESCRIPTION
**Pull request by delphinus**
Neovim now has Lua API to define autocmd (https://github.com/neovim/neovim/pull/14661). This PR makes it use them.

### CODE REVIEW

File: lua/agrp.lua [hunk 1]
```diff
@@ -1,75 +1,64 @@
...
+        opt.nested = definition[1].nested and true or false
+        opt.pattern = definition[3]
+        event = definition[2]
+        cb_or_cmd = definition[4]
       else
-        error'each definition should have 3 values (+options (once, nested))'
+        error "each definition should have 3 values (+options (once, nested))"
       end
     else
       for _, d in ipairs(definition) do
```
Comment (mendes-davi) [inline lua/agrp.lua:63]: in this loop, each definition `d` is not being registered as an autocmd only the last one.

**Review by mendes-davi — COMMENTED** [review]
hi, i've noticed a little issue when `key` is a table containing multiple autocmd definitions.

Thank you for your work!

File: lua/agrp.lua (hunk 1 above)
Comment (delphinus) [reply lua/agrp.lua:63]: Thank you for your review! I clean up logic to call `nvim_create_autocmd()`.

**Comment by delphinus** [comment]
Neovim v0.7 released! This PR should be merged.

**Pull request merged**